### ISSUE TITLE
False positive in unused_lifetimes lint

### ISSUE THREAD

**Issue by fhartwig** [Manishearth/rust-clippy#550]
Given the following function: 
```
impl Foo {
    fn bar<'a, 'b>(&'a self, _arg: &'b str) -> &'b str { unimplemented!() }
}
```
clippy complains that

```
warning: this lifetime isn't used in the function definition, #[warn(unused_lifetimes)] on by default
src/lib.rs:23     fn bar<'a, 'b>(&'a self, _arg: &'b str) -> &'b str { unimplemented!() }
```
Looks like lifetimes on `&self` arguments aren't handled correctly.

**Comment by fhartwig** [comment]
I think I know how to fix this, hope to send a PR for this later.
